tiledb/sm/query/reader_base.cc

```diff
@@ -414,228 +414,200 @@ Status ReaderBase::read_tiles(
   if (result_tiles->empty())
     return Status::Ok();
 
-  // Reading tiles are thread safe. However, we will perform
-  // them on this thread if there is only one read to perform.
-  if (names->size() == 1) {
-    RETURN_NOT_OK(read_tiles(names->at(0), result_tiles));
-  } else {
-    const auto status = parallel_for(
-        storage_manager_->compute_tp(),
-        0,
-        names->size(),
-        [&](const uint64_t i) {
-          RETURN_NOT_OK(read_tiles(names->at(i), result_tiles));
-          return Status::Ok();
-        });
-
-    RETURN_NOT_OK(status);
-  }
-
-  return Status::Ok();
-}
-
-Status ReaderBase::read_tiles(
-    const std::string& name,
-    const std::vector<ResultTile*>* result_tiles) const {
-  // Shortcut for empty tile vec
-  if (result_tiles->empty())
-    return Status::Ok();
-
-  // Read the tiles asynchronously
-  std::vector<ThreadPool::Task> tasks;
-  RETURN_CANCEL_OR_ERROR(read_tiles(name, result_tiles, &tasks));
-
-  // Wait for the reads to finish and check statuses.
-  auto statuses = storage_manager_->io_tp()->wait_all_status(tasks);
-  for (const auto& st : statuses)
-    RETURN_CANCEL_OR_ERROR(st);
-
-  return Status::Ok();
-}
-
-Status ReaderBase::read_tiles(
-    const std::string& name,
-    const std::vector<ResultTile*>* result_tiles,
-    std::vector<ThreadPool::Task>* const tasks) const {
-  // Shortcut for empty tile vec
-  if (result_tiles->empty())
-    return Status::Ok();
-
-  // For each tile, read from its fragment.
-  const bool var_size = array_schema_->var_size(name);
-  const bool nullable = array_schema_->is_nullable(name);
+  // Populate the list of regions per file to be read.
+  std::map<URI, std::vector<std::tuple<uint64_t, void*, uint64_t>>> all_regions;
```

Comment:
Do we want a `map` or `unordered_map`?

Also cc: @lums658 another place where a concurrent map/unordered map data structure would simplify things.

Reply:
@Seth I'm not sure as I didn't look into VFS and I didn't create the original code.

Reply:
I think `unordered_map` should do here. I don't think order on `URI` matters.

Reply:
Slightly worried this might introduce some perf problems so filing a separate ticket.
https://app.shortcut.com/tiledb-inc/story/12331/move-from-map-to-unordered-map-in-read-tiles-for-all-regions